storagedriver/azure/azure.go

```diff
@@ -174,11 +193,17 @@ func (d *Driver) WriteStream(path string, offset, size int64, reader io.ReadClos
 		}
 	}
 
-	if offset != resumableOffset {
-		return storagedriver.InvalidOffsetError{Path: path, Offset: offset}
+	if offset < resumableOffset {
```

Comment:
Yep, we discussed this with @stevvooe early on. For now we'll just support append-only. In case the use case changes and random writes need to work, we can refactor the code. It will require a much complex logic to download & split & rearrange the uploaded blocks etc.

If it's not really the actual use case, I favor not implementing it. :)